src/screens/Ticketing/Ticket/Details/QrCode.tsx

```diff
@@ -5,39 +5,76 @@ import {TicketTexts, useTranslation} from '@atb/translations';
 import {SvgXml} from 'react-native-svg';
 import React, {useCallback, useEffect, useState} from 'react';
 import {StyleSheet, useTheme} from '@atb/theme';
-import {useMobileTokenContextState} from '@atb/mobile-token/MobileTokenContext';
+import {
+  useHasEnabledMobileToken,
+  useMobileTokenContextState,
+} from '@atb/mobile-token/MobileTokenContext';
 import qrcode from 'qrcode';
 import useInterval from '@atb/utils/use-interval';
 import MessageBox from '@atb/components/message-box';
 import ThemeText from '@atb/components/text';
-import {findInspectable, getDeviceName} from '@atb/mobile-token/utils';
-import {RemoteToken} from '@atb/mobile-token/types';
+import {
+  findInspectable,
+  getDeviceName,
+  isTravelCardToken,
+} from '@atb/mobile-token/utils';
+import {FareContract} from '@atb/tickets';
 
 type Props = {
   validityStatus: ValidityStatus;
+  isInspectable: boolean;
+  fc: FareContract;
 };
 
-export default function QrCode({validityStatus}: Props) {
-  const {remoteTokens, deviceIsInspectable, isLoading, isError, retry} =
+export default function QrCode({
+  validityStatus,
+  isInspectable,
+  fc,
+}: Props): JSX.Element | null {
+  const status = useQrCodeStatus(validityStatus, isInspectable);
+
+  switch (status) {
+    case 'none':
+      return null;
+    case 'loading':
+      return <LoadingQr />;
+    case 'static':
+      return <StaticQr fc={fc} />;
+    case 'mobiletoken':
+      return <MobileTokenQr fc={fc} />;
+    case 'other':
+      return <DeviceNotInspectable />;
+  }
+}
+
+const useQrCodeStatus = (
+  validityStatus: ValidityStatus,
+  isInspectable: boolean,
+) => {
+  const {remoteTokens, deviceIsInspectable, isLoading, isError} =
     useMobileTokenContextState();
+  const mobileTokenEnabled = useHasEnabledMobileToken();
 
-  if (validityStatus !== 'valid') return null;
+  if (!isInspectable) return 'none';
+  if (validityStatus !== 'valid') return 'none';
 
-  if (isLoading) return <QrCodeLoading />;
-  if (isError) return <QrCodeError retry={retry} />;
-  if (deviceIsInspectable) return <QrCodeSvg />;
+  if (!mobileTokenEnabled) return 'static';
 
-  const inspectableToken = findInspectable(remoteTokens);
-  if (inspectableToken) {
-    return <QrCodeDeviceNotInspectable inspectableToken={inspectableToken} />;
-  }
+  if (isLoading) return 'loading';
+  if (isError) return 'static';
+  if (deviceIsInspectable) return 'mobiletoken';
```

Comment:
After seeing so many conditions it has become a little difficult to differentiate between several terminologies like isInspectable and deviceIsInspectable? Not sure if its just me 😅

Reply:
sounds much clearer :)